Applications/Spire/Include/Spire/Ui/CurrencyComboBox.hpp

```diff
@@ -1,21 +1,16 @@
 #ifndef SPIRE_CURRENCY_COMBO_BOX_HPP
 #define SPIRE_CURRENCY_COMBO_BOX_HPP
-#include <QWidget>
+#include <QLineEdit>
 #include "Nexus/Definitions/Currency.hpp"
 #include "Spire/Spire/Spire.hpp"
+#include "Spire/Ui/StaticDropDownMenu.hpp"
 
 namespace Spire {
 
   //! Represents a combo box for selecting Currencies.
-  class CurrencyComboBox : public QWidget {
+  class CurrencyComboBox : public QLineEdit {
```

Comment:
Inheriting from ```QLineEdit``` is not a good idea since that requires ```CurrencyComboBox``` to broaden the functionality provided by a ```QLineEdit``` when the goal is to restrict that functionality (ie. it's a violation of the Liskov substitution principle).

Restricting functionality is done with encapsulation, that is ```CurrencyComboBox``` can have a ```QLineEdit``` as a data member and have full control over it.

Reply:
I don't think it's possible to have the same affect with the QLineEdit as a member, because the sender of the signal would be the member QLineEdit, and not the combo box itself.

Reply:
What effect are you referring to?

Reply:
This Qt signal usage applies only to the combo boxes' use as a cell editor, but the desire is to be able to use a Qt signal to signal when the user has selected a value and call a slot in the delegate. The delegate's createEditor function is const, so the Qt signal/slot system provides a way to make that connection without modifying any of the delegates members, like we would have to do when storing a boost::signals2::connection.

By having the QLineEdit be a member of the combo box, the delegate would identify the sender of the editingFinished signal as the member QLineEdit, so it wouldn't be able to access the combo box's current item.

There's workarounds for this, since this is done in the same delegate we would always know that the parent to the QLineEdit sender is a combo box, but ideally the combo box would have the signal itself.